### ISSUE TITLE
Inconsistent behavior of `attachment.attachment_type` in legacy vs new Outlook

### ISSUE THREAD

**Issue by agroebe** [O365/python-o365#1060]
Hello,

I'm observing inconsistent behavior of `attachment.attachment_type` in the legacy vs new versions of Outlook. I am on the latest version of Python O365 (2.0.34)

In legacy Outlook, my message attachment is properly being classified as `attachment_type = "item"`

However, when I attach the exact SAME message attachment using NEW Outlook, it is being classified as `attachment_type == 'file'` no matter what it is I attach.

Basically, I am following the exact same steps in legacy vs new Outlook and observing different results. I am assuming this has something to do with how this library parses the attachment to determine its type.

Thanks!

**Comment by agroebe** [comment]
I was able to call `to_api_data()` on the attachment and find out that the API seems to consider the attachment to have the following data type:

```
"@odata.type": '#microsoft.graph.fileAttachment'
```

Looks like this is causing the item check to fail, thus it falls back to a `file` attachment...

**Comment by alejcas** [comment]
Any proposal on how to solve this?

Thanks

**Comment by agroebe** [comment]
> Any proposal on how to solve this?
> 
> Thanks

Not particularly. If the O365 API is classifying the attachment type as a file, I'm not sure what you can do about it. Again, this is only affecting the new version of Outlook, which will hopefully have this issue fixed by the time the legacy version is deprecated.
